packages/discord.js/src/structures/client/BaseClient.ts

```diff
@@ -0,0 +1,73 @@
+import EventEmitter from 'events';
+import { ClientOptions } from '../../typedefs/ClientOptions';
+export class BaseClient extends EventEmitter {
+	public constructor(options?: ClientOptions) {
+		super();
+		this.options = options;
+	}
+	// private readonly api
+	// private rest
+
+	private readonly _timeouts = new Set<NodeJS.Timeout>();
+	private readonly _intervals = new Set<NodeJS.Timeout>();
+	private readonly _immediates = new Set<NodeJS.Immediate>();
+	public options: ClientOptions | undefined;
```

Comment:
This field can be inlined.
```ts
public constructor(public options?: ClientOptions) {
	super();
}
```

Reply:
I thought @kyranet said not to in-line the fields? I thought we were keeping them separate

Reply:
Oh okay sorry.

Reply:
All good :)